dask/dataframe/groupby.py

```diff
@@ -1456,6 +1453,11 @@ def __init__(
         if observed is not None:
             self.observed["observed"] = observed
 
+        if isinstance(self, DataFrameGroupBy):
+            all_dtypes = self.obj._meta.dtypes
+            by_names = [x.name if isinstance(x, Series) else x for x in by_items]
+            self._result_dtypes = all_dtypes.drop(by_names, errors="ignore")
```

Comment:
Looks like I still need this `ignore` here, otherwise these two tests fail:

```
dask/dataframe/tests/test_groupby.py::test_dataframe_aggregations_multilevel
dask/dataframe/tests/test_groupby.py::test_groupby_column_and_index_agg_funcs
```

Looks like it happens when trying to drop a name that belongs to an index, vs a column. Such a name would not be part of columns on which agg function is performed.